src/Abstractions/RetryPolicy.cs

```diff
@@ -123,11 +133,56 @@ public class RetryPolicy
     /// </value>
     public TimeSpan RetryTimeout { get; }
 
+    /// <summary>
+    /// Gets a delegate to call on exception to determine if retries should proceed.
+    /// </summary>
+    /// <value>
+    /// Defaults delegate that always returns true (i.e., all exceptions are retried).
+    /// </value>
+    public Func<Exception, bool> Handle { get; private set; }
```

Comment:
I'd rather we used the existing `HandleAsync`, but I understand the core options don't support async retry. But we _could_ support it - we just can't convert to core options anymore and need to do the retry entirely in this library. Which isn't the worst, it is just a fancy loop helper.

@cgillum what are your thoughts on sync and use core retry options, or async and implement it entirely here?

Reply:
> what are your thoughts on sync and use core retry options, or async and implement it entirely here?

For this PR, I suggest we stick with sync and use core retry options to meet the current need. But I'd be open to reconsidering this as part of a larger refactor at some point to get rid of the DurableTask.Core dependency entirely.

Reply:
Sounds good - my only concern then is the ctor breaking change.